src/H5Pencdec.c

```diff
@@ -333,7 +333,7 @@ H5P__encode_cb(H5P_genprop_t *prop, void *_udata)
         /* Encode (or not, if the 'encode' flag is off) the property's name */
         prop_name_len = HDstrlen(prop->name) + 1;
         if (udata->encode) {
-            HDstrncpy((char *)*(udata->pp), prop->name, prop_name_len);
```

Comment:
Again, we don't know the size of the output buffer, so using strcpy here is no less safe than strncpy and strncpy gives a warning since we're just giving it the length of the source being copied.